packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -1392,4 +1381,40 @@ class CupertinoDialogRoute<T> extends RawDialogRoute<T> {
         barrierLabel: barrierLabel ?? CupertinoLocalizations.of(context).modalBarrierDismissLabel,
         barrierColor: barrierColor ?? CupertinoDynamicColor.resolve(kCupertinoModalBarrierColor, context),
       );
+
+  CurvedAnimation? _fadeAnimation;
+
+  void _setAnimation(Animation<double> animation) {
+    if (_fadeAnimation?.parent.hashCode != animation.hashCode) {
+      _fadeAnimation?.dispose();
```

Comment:
why hashcode, not direct equality?

Reply:
copy-pasted from #147816 Was added while debugging issue, its fixed now.